File: doma-core/src/main/java/org/seasar/doma/jdbc/criteria/declaration/DeclarationSupport.java
package org.seasar.doma.jdbc.criteria.declaration;

import org.seasar.doma.jdbc.criteria.context.Operand;
import org.seasar.doma.jdbc.criteria.def.PropertyDef;

public class DeclarationSupport {

  public <PROPERTY> Operand.Prop toProp(final PropertyDef<PROPERTY> propertyDef) {
    return new Operand.Prop(propertyDef);
  }

  public <PROPERTY> Operand.Param toParam(
      final PropertyDef<PROPERTY> propertyDef, final PROPERTY value) {
    return new Operand.Param(propertyDef, value);
  }
}

File: doma-criteria/src/main/kotlin/org/seasar/doma/criteria/declaration/DeclarationSupport.kt
package org.seasar.doma.criteria.declaration

import org.seasar.doma.criteria.context.Operand
import org.seasar.doma.def.PropertyDef
import org.seasar.doma.internal.jdbc.scalar.Scalars
import org.seasar.doma.internal.jdbc.sql.ScalarInParameter
import org.seasar.doma.jdbc.Config

class DeclarationSupport(private val config: Config) {

    fun <PROPERTY> toProp(propDef: PropertyDef<PROPERTY>): Operand.Prop {
        return Operand.Prop(propDef)
    }

    fun <PROPERTY> toParam(propDef: PropertyDef<PROPERTY>, value: PROPERTY?): Operand.Param {
        val clazz = propDef.asClass()
        val supplier = Scalars.wrap(value, clazz, false, config.classHelper)
        return Operand.Param(ScalarInParameter(supplier.get()))
    }
}
